pkg/apis/seedmanagement/types_managedseed.go

```diff
@@ -0,0 +1,182 @@
+type Gardenlet struct {
+	// Deployment specifies certain gardenlet deployment parameters, such as the number of replicas,
+	// the image, which bootstrap mechanism to use (bootstrap token / service account), etc.
+	Deployment *GardenletDeployment
+	// Config is the GardenletConfiguration used to configure gardenlet.
+	Config *gardenletconfigv1alpha1.GardenletConfiguration
+	// Bootstrap is the mechanism that should be used for bootstrapping gardenlet connection to the Garden cluster. One of ServiceAccount, Token.
+	// If specified, a service account or a bootstrap token will be created in the garden cluster and used to compute the bootstrap kubeconfig.
+	// If not specified, the gardenClientConnection.kubeconfig field will be used to connect to the Garden cluster.
+	GardenConnectionBootstrap *GardenConnectionBootstrap
+	// SeedConnection is the mechanism for gardenlet connection to the Seed cluster. Must equal ServiceAccount if specified.
+	// If not specified, the seedClientConnection.kubeconfig field will be used to connect to the Seed cluster.
+	SeedConnection *SeedConnection
+	// MergeParentConfig specifies whether the deployment parameters and GardenletConfiguration of the parent gardenlet
+	// should be merged with the specified deployment parameters and GardenletConfiguration. Defaults to false.
+	MergeParentConfig bool
+}
+
+// GardenletDeployment specifies certain gardenlet deployment parameters, such as the number of replicas,
+// the image, which bootstrap mechanism to use (bootstrap token / service account), etc.
+type GardenletDeployment struct {
+	// ReplicaCount is the number of gardenlet replicas. Defaults to 1.
+	ReplicaCount *int32
+	// RevisionHistoryLimit is the number of old gardenlet ReplicaSets to retain to allow rollback. Defaults to 10.
+	RevisionHistoryLimit *int32
+	// ServiceAccountName is the name of the ServiceAccount to use to run gardenlet pods.
+	ServiceAccountName string
+	// Image is the gardenlet container image.
+	Image *Image
+	// Resources are the compute resources required by the gardenlet container.
+	Resources *corev1.ResourceRequirements
+	// PodLabels are the labels on gardenlet pods.
+	PodLabels map[string]string
+	// PodAnnotations are the annotations on gardenlet pods.
+	PodAnnotations map[string]string
+	// AdditionalVolumes is the list of additional volumes that should be mounted by gardenlet containers.
+	AdditionalVolumes []corev1.Volume
+	// AdditionalVolumeMounts is the list of additional pod volumes to mount into the gardenlet container's filesystem.
+	AdditionalVolumeMounts []corev1.VolumeMount
+	// Env is the list of environment variables to set in the gardenlet container.
+	Env []corev1.EnvVar
+	// VPA specifies whether to enable VPA for gardenlet. Defaults to false.
+	VPA *bool
+	// ImageVectorOverwrite is the gardenlet image vector overwrite.
+	// More info: https://github.com/gardener/gardener/blob/master/docs/deployment/image_vector.md.
+	ImageVectorOverwrite string
```

Comment:
I would suggest to move the ImageVector go types from `pkg/utils/imagevector/types.go` to some API package and reuse them here, so we can have strongly typed fields and proper validation (instead of plain strings).

Reply:
I would prefer to have a very close mapping between the fields used here and the values that can be specified in gardenlet's `values.yaml`. Could this be done later?

Reply:
I would also vote for keeping it for now.

Reply:
> Could this be done later?

I guess, we would need a new API version for that, because changing the type later on would be an incompatible change, right? 
If we merge it like this, we will definitely need to change the API later on, which I would try to avoid, if we already know, that it will be coming.

We can also move the ImageVector types into an API package in a different PR and then rebase this one, if the concern is only about not making this PR too large.

Reply:
@timebertt @rfranzke Is this even a good idea? I mean, with the `imageVectorOverwrite` settings we are basically specifying the contents of a file, to be written via the Helm chart and used by totally different components. Are you sure you want to make this format an API, with all the resulting implications? 

Even if it requires an API version change later on, I would still try to avoid it here because I see such a change beyond the scope of what we are trying to achieve with this PR. From my perspective it requires a separate decision that should then be carried out with all the resulting implications.

Reply:
Hmm, now that I'm thinking about this again, we could even go without those fields for the time being and safe the headaches.
Even with the ManagedSeed in place, the landscape owner will have to install at least one gardenlet manually (including the desired image overwrites).
We could then always reuse the image overwrites from the parent gardenlet for reconciling `ManagedSeeds`.
Probably, there is no immediate use case for using different overwrites per `ManagedSeed` as of now, as this also hasn't been supported before.

TL;DR: If I didn't miss anything, I would vote for dropping the fields in the first version of the resource.
WDYT?

Reply:
OK, I will drop them.